src/linker/pipeline.py

```diff
@@ -25,72 +21,36 @@ def set_runner(self, runner: Callable) -> None:
     def run(self, results_dir: Path):
         if not self.runner:
             raise RuntimeError("Runner has not been set.")
-        for step_name in self.steps:
-            self.runner(
+        for step in self.steps:
+            step.run(
+                runner=self.runner,
                 container_engine=self.config.container_engine,
                 input_data=self.config.input_data,
                 results_dir=results_dir,
-                step_name=step_name,
-                implementation_dir=self._get_implementation_directory(step_name),
-                container_full_stem=self._get_container_full_stem(step_name),
             )
 
     #################
     # Setup methods #
     #################
 
-    def _get_steps(self) -> Tuple:
+    def _get_steps(self) -> Tuple[Step, ...]:
         spec_steps = tuple(self.config.pipeline["steps"])
-        steps = tuple([x for x in spec_steps if x in STEPS])
-        unknown_steps = [x for x in spec_steps if x not in STEPS]
+        # TODO: The order is defined below but should be part of the
+        ## PipelineSchema when that becomes a thing
+        steps = tuple(
+            Step(step, self.config) for step in Step.IMPLEMENTATIONS if step in spec_steps
+        )
+        unknown_steps = [step for step in spec_steps if step not in Step.IMPLEMENTATIONS]
```

Comment:
Out of scope for this PR, but I think this check could be simpler. The set of steps in the pipeline specification should _exactly_ match the steps that the Pipeline knows about (at least until we get to hierarchically nested sub-steps).